diffrax/misc/discontinuity.py

```diff
@@ -0,0 +1,62 @@
+from typing import Callable
+
+import diffrax
+import jax
+import jax.numpy as jnp
+from diffrax.custom_types import Array, Scalar
+
+
+def find_root_for_one_delay_one_discontinuity(
+    delay: Callable,
+    discontinuity: float,
+    t0: float,
+    t1: float,
+    y0: Array,
+    y1: Array,
+    atol: float = 1e-6,
+    rtol: float = 1e-3,
+    max_steps: int = 10,
+):
+    cubic_coeffs = diffrax.backward_hermite_coefficients(
+        jnp.array([t0, t1]), jnp.array([y0, y1])
+    )
+    interpol = diffrax.CubicInterpolation(ts=jnp.array([t0, t1]), coeffs=cubic_coeffs)
+    g = lambda t, args: t - delay(t, interpol.evaluate(t), args) - discontinuity
+    root_solver = diffrax.NewtonNonlinearSolver(
+        atol=atol, rtol=rtol, max_steps=max_steps
+    )
+    pred = g(t0, None) * g(t1, None) > 0
+
+    def true_fun(_):
+        return jnp.nan
+
+    def false_fun(t0):
+        # Since steps are pretty small the root finding process
+        # should be almost certainly successful hence only taking the root argument
+        return root_solver(g, t0, None).root
+
+    return jax.lax.cond(pred, true_fun, false_fun, t0)
```

Comment:
Obviously I've put together a replacement code snippet above, but for completeness:
- Don't use `jnp.nan`. For placeholder values Diffrax uses `jnp.inf` instead. This means we can debug using `JAX_DEBUG_NANS=1` without getting false positives.
- I try to avoid introducing default arguments in internal APIs.
- You've hardcoded `args=True`.